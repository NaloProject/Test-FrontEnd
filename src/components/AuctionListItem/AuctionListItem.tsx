import { FC } from 'react'
import { Sale } from '@Redux/types'

import { ReactComponent as FlameIcon } from '@Svg/flame-icon.svg'

import Image from 'next/image'
import Link from 'next/link'

export type AuctionListItemProps = {
  auction: Sale
}

const AuctionListItem: FC<AuctionListItemProps> = ({ auction }) => (
  <Link
    href={`/auction/${auction.id}`}
    key={auction.id}
    className="flex w-full mb-4 pb-4 border border-t-0 border-r-0 border-b-1 sm:border-b-0 border-b-slate-200 dark:border-b-slate-700 border-l-0 last-of-type:border-b-0"
  >
    <article className="flex w-full">
      <div className="h-[64px] w-[64px] lg:h-[128px] lg:w-[128px] mr-4">
        <div className="overflow-hidden flex rounded-lg w-full relative h-[64px] w-[64px] lg:h-[128px] lg:w-[128px]">
          <Image
            src={auction.image}
            alt={auction.name}
            height={64}
            width={64}
            className="pointer-events-none h-[64px] w-[64px] lg:h-[128px] lg:w-[128px]"
          />
        </div>
      </div>

      <div className="flex-1">
        <h3 className="font-medium lg:font-bold text-[16px] lg:text-[24px] text-slate-700 dark:text-slate-300 mb-1">
          {auction.name}
        </h3>
        <h4 className="text-[12px] text-slate-300 dark:text-slate-600">
          {auction.totalMinted} Editions minted
        </h4>
      </div>

      <div className="flex flex-col items-end">
        <span className="font-bold text-[14px] text-slate-700 dark:text-slate-300 mb-1">
          {auction.price} ETH
        </span>
        <span className="text-[13px] text-slate-400 flex-1">Starting Bid</span>
        {auction.isHot ? (
          <div className="hidden lg:block bg-slate-200 dark:bg-slate-400 bottom-[20px] flex rounded-lg px-3 py-2">
            <span className="text-slate-700 font-medium mr-2 text-[12px]">
              {auction.timeLeft} left
            </span>

            <span className="inline-block w-[12px] h-[12px]">
              <FlameIcon />
            </span>
          </div>
        ) : null}
      </div>
    </article>
  </Link>
)

export default AuctionListItem
