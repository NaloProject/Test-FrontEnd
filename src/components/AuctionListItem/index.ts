import AuctionListItem from './AuctionListItem'

export default AuctionListItem
