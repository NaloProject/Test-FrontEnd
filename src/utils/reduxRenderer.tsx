import { configureStore, EnhancedStore } from '@reduxjs/toolkit'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ReactElement, FC, PropsWithChildren } from 'react'
import { Provider } from 'react-redux'

import { combined, RootState } from '@Redux/store'
import INITIAL_STATE from './initialState'

type RenderWithReduxOptions = Omit<RenderOptions, 'queries'> & {
  initialState?: RootState
  store?: EnhancedStore
}

export const renderWithRedux = (
  ui: ReactElement,
  {
    initialState = INITIAL_STATE,
    store = configureStore({
      middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
          immutableCheck: false,
          serializableCheck: false,
        }),
      preloadedState: initialState,
      reducer: combined,
    }),
    ...renderOptions
  }: RenderWithReduxOptions = {}
): RenderResult => {
  const Wrapper: FC<PropsWithChildren> = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  )

  return render(ui, { wrapper: Wrapper, ...renderOptions })
}
