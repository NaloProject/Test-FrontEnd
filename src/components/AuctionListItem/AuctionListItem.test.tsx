import { renderWithRedux } from '@Utils/reduxRenderer'
import { auctionLisItemStub } from './AuctionListItem.stub'
import AuctionListItem from './AuctionListItem'

describe('AuctionListItem', () => {
  it('should mount first', async () => {
    const { container } = renderWithRedux(
      <AuctionListItem auction={auctionLisItemStub} />
    )

    expect(container).toMatchSnapshot()
  })
})
