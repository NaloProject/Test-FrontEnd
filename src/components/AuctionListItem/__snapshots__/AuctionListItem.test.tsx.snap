// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuctionListItem should mount first 1`] = `
<div>
  <a
    class="flex w-full mb-4 pb-4 border border-t-0 border-r-0 border-b-1 sm:border-b-0 border-b-slate-200 dark:border-b-slate-700 border-l-0 last-of-type:border-b-0"
    href="/auction/6"
  >
    <article
      class="flex w-full"
    >
      <div
        class="h-[64px] w-[64px] lg:h-[128px] lg:w-[128px] mr-4"
      >
        <div
          class="overflow-hidden flex rounded-lg w-full relative h-[64px] w-[64px] lg:h-[128px] lg:w-[128px]"
        >
          <img
            alt="Ceres Searching"
            class="pointer-events-none h-[64px] w-[64px] lg:h-[128px] lg:w-[128px]"
            data-nimg="1"
            decoding="async"
            height="64"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fntfs%2FContemporary-Sculptures-For-The-Home.jpg&w=128&q=75"
            srcset="/_next/image?url=%2Fimages%2Fntfs%2FContemporary-Sculptures-For-The-Home.jpg&w=64&q=75 1x, /_next/image?url=%2Fimages%2Fntfs%2FContemporary-Sculptures-For-The-Home.jpg&w=128&q=75 2x"
            style="color: transparent;"
            width="64"
          />
        </div>
      </div>
      <div
        class="flex-1"
      >
        <h3
          class="font-medium lg:font-bold text-[16px] lg:text-[24px] text-slate-700 dark:text-slate-300 mb-1"
        >
          Ceres Searching
        </h3>
        <h4
          class="text-[12px] text-slate-300 dark:text-slate-600"
        >
          2
           Editions minted
        </h4>
      </div>
      <div
        class="flex flex-col items-end"
      >
        <span
          class="font-bold text-[14px] text-slate-700 dark:text-slate-300 mb-1"
        >
          10.89
           ETH
        </span>
        <span
          class="text-[13px] text-slate-400 flex-1"
        >
          Starting Bid
        </span>
        <div
          class="hidden lg:block bg-slate-200 dark:bg-slate-400 bottom-[20px] flex rounded-lg px-3 py-2"
        >
          <span
            class="text-slate-700 font-medium mr-2 text-[12px]"
          >
            2 days
             left
          </span>
          <span
            class="inline-block w-[12px] h-[12px]"
          >
            <svg />
          </span>
        </div>
      </div>
    </article>
  </a>
</div>
`;
